bundles/org.eclipse.equinox.p2.transport.ecf/src/org/eclipse/equinox/internal/p2/transport/ecf/RepositoryTransport.java

```diff
@@ -296,7 +296,7 @@ public static DownloadStatus forException(Throwable t, URI toDownload) {
 			String value = System.getProperty(TIMEOUT_RETRY);
 			if (value != null) {
 				try {
-					int retry = Integer.valueOf(value).intValue();
+					int retry = Integer.parseInt(value);
```

Comment:
I think one even can use here `java.lang.Integer.getInteger`

Reply:
Is it any better?

Reply:
As it combines checking for a system property with parsing the integer the whole block could be simplified, I also would assume that the JDK always choose the most performant way to parse string > int.

Reply:
Thanks. So yes, let's use `Integer.getInteger()` instead.